multi-agent/lib/mas/elements/llms/google_genai/google_genai.py

```
@@ -75,19 +102,46 @@ def stream(
+                    level="ERROR",
+                    status_message=f"Google GenAI API error: {type(e).__name__}: {e}",
+                )
+                raise
+            for chunk in stream_iter:
+                if chunk.text:
+                    accumulated_text += chunk.text
+                    yield chunk.text
+
+                for part in (chunk.parts or []):
+                    if part.function_call is not None:
+                        collected_parts.append(part)
```

Ruff (0.15.21)

[warning] Use `list.extend` to create a transformed list

(PERF401)